driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocumentCodec.java

```diff
@@ -59,7 +60,7 @@
                 .register(changeStreamDocumentClassModel)
                 .build();
 
-        CodecRegistry registry = fromRegistries(fromProviders(provider), codecRegistry);
+        CodecRegistry registry = fromRegistries(fromProviders(provider, new BsonValueCodecProvider()), codecRegistry);
```

Comment:
I think you can remove the `.register(MongoNamespace.class)` above as thats no longer needed.

Interesting using `new BsonValueCodecProvider()`? Previously the code was manually setting the codecs for BsonTypes but as custom mapping of BsonTypes in a BsonDocument isn't possible (it is with Document) I think we can just use this approach. So you can delete the codecs being set for clusterTime, documentKey and resumeToken above.

Reply:
This was necessary because otherwise it couldn't find a codec for the UpdateDescription#updatedFields property (which is of type BsonDocument) unless the application's registry contained one.

Reply:
When I remove `.register(MongoNamespace.class)`, the "should decode <operation-type> to ChangeStreamDocument" tests all fail. An exception also gets thrown, that I've attached. When I follow the stack trace, it seems like "namespace" is getting added to the propertyModels in ClassModel. Is this a bug?
![Exception Stack Trace](https://user-images.githubusercontent.com/12755364/45644799-9868e880-ba8c-11e8-8834-1e253d069b12.png)

Reply:
Ah that is an issue! We don't need to encode that getter. Added a comment in the Pojo.

Reply:
Done